app/main.py

```diff
@@ -1,13 +1,55 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    formatted_errors = [
+        {
+            "line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+        }
+        for error in errors
+    ]
+
+    status = "failed" if errors else "passed"
+
+    return {
+        "errors": formatted_errors,
+        "path": file_path,
+        "status": status
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    formatted_report = []
+
+    for file_path, errors in linter_report.items():
+        formatted_errors = [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in errors
+        ]
```

Comment:
This block of code is identical to the one in 'format_single_linter_file'. Consider refactoring to use a helper function to format errors, which can be reused in both functions.